src/groovy/au/org/emii/portal/proxying/RequestProxyingController.groovy

```diff
@@ -32,7 +32,7 @@ abstract class RequestProxyingController {
         }
         else if (!hostVerifier.allowedHost(url)) {
             log.info "Proxy: The url $url was not allowed"
-            render text: "Host for address '$url' not allowed", contentType: "text/html", encoding: "UTF-8", status: HTTP_400_BAD_REQUEST
+            render text: "Host for address '$url' not allowed", contentType: "text/html", encoding: "UTF-8", status: HTTP_403_FORBIDDEN
```

Comment:
It might be that if you return 403 you might not have a chance to actually set the text etc. I think verify it and just strip all of that if so - and return just 403 as you do.